tests/foreman/cli/test_leapp_client.py

```diff
@@ -0,0 +1,203 @@
+"""Test for LEAPP cli
+
+:Requirement: leapp
+
+:CaseLevel: Acceptance
+
+:CaseComponent: Leappintegration
+
+:Team: Rocket
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:CaseAutomation: Automated
+
+:Upstream: No
+"""
+import pytest
+from wrapanapi.entities.vm import VmState
+
+from robottelo.constants import PRDS
+
+CUSTOM_REPOSET = {
+    'rhel8_8_bos': 'Red Hat Enterprise Linux 8 for x86_64 - BaseOS (RPMs)',
+    'rhel8_8_aps': 'Red Hat Enterprise Linux 8 for x86_64 - AppStream (RPMs)',
+    'rhel9_2_bos': 'Red Hat Enterprise Linux 9 for x86_64 - BaseOS (RPMs)',
+    'rhel9_2_aps': 'Red Hat Enterprise Linux 9 for x86_64 - AppStream (RPMs)',
+}
+CUSTOM_REPOS = {
+    'rhel8_8_bos': {
+        'id': 'rhel-8-for-x86_64-baseos-rpms',
+        'name': 'Red Hat Enterprise Linux 8 for x86_64 - BaseOS RPMs 8.8',
+        'releasever': '8.8',
+    },
+    'rhel8_8_aps': {
+        'id': 'rhel-8-for-x86_64-appstream-rpms',
+        'name': 'Red Hat Enterprise Linux 8 for x86_64 - AppStream RPMs 8.8',
+        'releasever': '8.8',
+    },
+    'rhel9_2_bos': {
+        'id': 'rhel-9-for-x86_64-baseos-rpms',
+        'name': 'Red Hat Enterprise Linux 9 for x86_64 - BaseOS RPMs 9.2',
+        'releasever': '9.2',
+    },
+    'rhel9_2_aps': {
+        'id': 'rhel-9-for-x86_64-appstream-rpms',
+        'name': 'Red Hat Enterprise Linux 9 for x86_64 - AppStream RPMs 9.2',
+        'releasever': '9.2',
+    },
+}
+
+
+@pytest.fixture(scope="module")
+def setup_env(module_target_sat, module_org):
+    """Creating essential things and returning in form of dictiory for use"""
+    # 1. Import a subscription manifest
+    module_target_sat.upload_manifest(module_org)
+    lc_env = module_target_sat.api.LifecycleEnvironment(organization=module_org).create()
+    c_view = module_target_sat.api.ContentView(organization=module_org).create()
+    return {
+        'organization': module_org,
+        'satellite': module_target_sat,
+        'environment': lc_env,
+        'contentview': c_view,
+    }
+
+
+@pytest.mark.no_containers
+@pytest.mark.rhel_ver_match('8')
+def test_upgrade_rhel8_to_rehl9(
+    setup_env,
+    rhel_contenthost,
+):
```

Comment:
I was initially confused by this suggestion before determining that upgrading from RHEL 8.9 to RHEL 9.3 will only be a supported upgrade path for six months per [1]. Since the supported upgrade path matrix for LEAPP is not exactly intuitive, I think it would be worth linking to one of the documentation sources for this (either [1] or [2]) in the test or module metadata.

[1] https://access.redhat.com/articles/4263361
[2] https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html-single/upgrading_from_rhel_8_to_rhel_9/index#con_supported-upgrade-paths_upgrading-from-rhel-8-to-rhel-9

Reply:
Thank you for the suggestion, may I know how I can include the above-supported path [2] in a test?
@synkd

Reply:
@vijaysawant That's a good question, but it's one that I don't have a definitive answer to. We don't have an established standard for this sort of thing in the Robottelo or Testimony documentation. My first thought is to include the link in the `:ExpectedResults:` section of the test function docstring with a note that the expected successful RHEL versions can be found in that documentation link. I'm open to other suggestions, though.